frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -420,3 +420,14 @@ export type ResourceLinkProps = {
 
 export type UseK8sModel = (groupVersionKind?: GroupVersionKind) => [K8sKind, boolean];
 export type UseK8sModels = () => [{ [key: string]: K8sKind }, boolean];
+
+export type PerspectiveDetectorProps = {
+  setActivePerspective: (perspective: string) => void;
+};
+
+export type PerspectiveType = string;
+
+export type UseActivePerspective = () => [
+  PerspectiveType,
+  React.Dispatch<React.SetStateAction<PerspectiveType>>,
+];
```

Comment:
Added the types as well.. not sure entirely what this is for, but this should work.